src/main/kotlin/no/nav/omsorgspenger/personopplysninger/RelasjonMediator.kt

```diff
@@ -0,0 +1,81 @@
+package no.nav.omsorgspenger.personopplysninger
+
+import kotlin.reflect.full.memberProperties
+import no.nav.omsorgspenger.personopplysninger.pdl.HentRelasjonPdlResponse
+import no.nav.omsorgspenger.personopplysninger.pdl.HentRelasjonPdlResponse.Relasjon
+import no.nav.omsorgspenger.personopplysninger.pdl.PdlClient
+import org.slf4j.LoggerFactory
+
+internal class RelasjonMediator(
+        internal val pdlClient: PdlClient
+) {
+    private val secureLogger = LoggerFactory.getLogger("tjenestekall")
+
+    suspend fun hentRelasjoner(identitetsnummer: String, til: Set<String>, correlationId: String): RelasjonsLøsningsMap {
+        val response = pdlClient.HentRelasjonInfo(til.plus(identitetsnummer), correlationId)
+
+        if (!response.errors.isNullOrEmpty()) {
+            secureLogger.error("Fann feil vid hent av data fra PDL: ", response.errors.toString())
+            throw IllegalStateException("Fann feil vid hent av data fra PDL")
+        }
+
+        var relasjonsListe = mutableListOf<Map<String, Any>>()
+
+        val søkersMatrikkelId = response.data.hentPersonBolk
+                .filter { it.ident == identitetsnummer && it.code == "ok" }
+                .first().person?.bostedsadresse?.first()?.matrikkeladresse?.matrikkelId.orEmpty()
+
+        response.data.hentPersonBolk
+                .filter { it.code == "ok" && it.ident != identitetsnummer && it.person != null }
+                .map { personBolk ->
+                    personBolk.håndterResponse(identitetsnummer, søkersMatrikkelId).let {
+                        if (!it.isNullOrEmpty()) relasjonsListe.add(it)
+                    }
+                }
+
+        return mapOf(RelasjonerKey to relasjonsListe)
+
+    }
+
+    private fun HentRelasjonPdlResponse.PersonBolk.håndterResponse(søkersIdentitetsnummer: String, søkersMatrikkelId: String): Map<String, Any> {
+        var resultat = mutableMapOf<String, Any>()
+
+        this.person!!.familierelasjoner
```

Comment:
Er det filtrer ut på noe tidspunkt filtrert bort hvor `this.person` og `relatertPerson.relatertPersonRolle` kan være null?